### PR TITLE
feat(mentions): show mentions ordered by how it matches the filter

### LINKED ISSUE #2543: Mentions autocomplete popup does not show matches sorted
examples

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #2543

### CODE REVIEW

File: ui/app/AppLayouts/Chat/ChatColumn/SuggestionBox.qml [hunk 1]
```diff
@@ -108,42 +115,60 @@ Rectangle {
...
+                    width: 32
+                    height: 32
+                    anchors.verticalCenter: parent.verticalCenter
+                    anchors.left: parent.left
+                    anchors.leftMargin: Style.current.smallPadding
+                    source: model.identicon
+                }
+
+                StyledText {
+                    text: model[container.property.find(p => !!model[p])]
```
Comment (richard-ramos) [inline ui/app/AppLayouts/Chat/ChatColumn/SuggestionBox.qml:154]: TIL!

Comment (richard-ramos) [reply ui/app/AppLayouts/Chat/ChatColumn/SuggestionBox.qml:154]: Ah! it was already part of the suggestionfilter code, but nevertheless, I did not know it was possible to get a model property like this

**Review by richard-ramos — APPROVED** [review]

**Review by iurimatias — CHANGES_REQUESTED** [review]
there is an issue with the selector, for e.g in this screenshot I type `@em` expecting to see Emizzle first, but instead the cursor is somewhere else, also it doesn't look the results are ordered
Screen Shot 2021-07-09 at 11 44 17 AM

**Comment by jrainville** [comment]
@iurimatias You issues should be fixed

**Pull request merged**